test/models/QueryStateModelTest.ts

```diff
@@ -1,46 +1,57 @@
+/// <reference path="../../lib/jasmine/index.d.ts" />
 import { QueryStateModel } from '../../src/models/QueryStateModel';
 import * as Mock from '../MockEnvironment';
 
 export function QueryStateModelTest() {
-  describe('QueryStateModel', function() {
+  describe('QueryStateModel', () => {
     var queryState: QueryStateModel;
     var div: HTMLDivElement;
     var env: Mock.IMockEnvironment;
 
-    beforeEach(function() {
+    beforeEach(() => {
       div = document.createElement('div');
       env = new Mock.MockEnvironmentBuilder().withRoot(div).build();
       queryState = new QueryStateModel(div, undefined);
     });
 
-    afterEach(function() {
+    afterEach(() => {
       div = null;
       env = null;
       queryState = null;
     });
 
-    it('can determine if a facet is active if one facet has selected values', function() {
+    it('can determine if a facet is active if one facet has selected values', () => {
       queryState.registerNewAttribute('f:@foobar', ['foo', 'bar']);
+      queryState.set('f:@foobar', ['not', 'default', 'value']);
       expect(queryState.atLeastOneFacetIsActive()).toBe(true);
     });
 
-    it('can determine if a facet is active if more than one facet has selected values', function() {
+    it('can determine if a facet is active if more than one facet has selected values', () => {
       queryState.registerNewAttribute('f:@foobar', ['foo', 'bar']);
       queryState.registerNewAttribute('f:@foobar2', ['foo2']);
+      queryState.set('f:@foobar', ['not', 'default']);
+      queryState.set('f:@foobar2', ['not', 'default', 'for', 'facet', '2']);
```

Comment:
A question: how does `.set()` work and are the strings in the array arbitrary or keywords that trigger special behavior? or both?

Reply:
In the query state, you can register any arbitrary key and register the default value for that key. They key is always a string, the default value can be anything.

So with `queryState.registerNewAttribute('f:@foobar', ['foo', 'bar'])` the default value for `f:@foobar` is an array with the value `foo`, and `bar`.

Then, when we call set, the query state will check if the new value we are setting is of the correct type when compared to the default value (an array of string).

The thing I modified in `QueryState` is the method `atLeastOneFacetIsActive` which was checking that there was at least one key registered for a facet (so, something that begins with `f:@some_field_name` ) and that it was not an empty array (because the default state for a "standard" facet is "no value selected").

This check was faulty for SliderFacet, for which the default value will instead be an array with  the minimum and maximum value possible in the index (for example : `@number_view = [0,1000]`.), and not an empty array.

So I modified the method to compare not against an empty array, but against an array that is different than the "default" value, which will works for both standard and slider facet.